google/cloud/forseti/common/util/date_time.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2017 The Forseti Security Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Common utility functions regarding date and time."""
+
+from datetime import datetime
+
+from google.cloud.forseti.common.util import logger
+from google.cloud.forseti.common.util import string_formats
+
+LOGGER = logger.get_logger(__name__)
+
+
+class UtilBaseDateTimeException(Exception):
+    """A Base UtilDateTime Exception."""
+
+
+class UtilDateTimeValueError(UtilBaseDateTimeException):
+    """Invalid Value Given for a Request."""
+
+
+class UtilDateTimeTypeError(UtilBaseDateTimeException):
+    """Type Error for a given Request."""
+
+
+def get_datetime_from_string(string, string_format):
+    """Get a datetime object from a string in the requested string_format.
+
+        Args:
+            string (str): The timestamp in the form of a string.
+            string_format (str): A string used for formatting.
+
+        Raises:
+           UtilDateTimeTypeError: When datetime.strptime() raises a TypeError.
+           UtilDateTimeValueError: When datetime.strptime() raises a ValueError.
+
+        Returns:
+            datetime: A datetime object as requested.
+    """
+    try:
+        result = datetime.strptime(string, string_format)
+    except TypeError:
+        LOGGER.error('Unable to create a datetime with %s in format %s',
+                     string, string_format)
+        raise UtilDateTimeTypeError
+    except ValueError:
+        LOGGER.error('Unable to create a datetime with %s in format %s',
+                     string, string_format)
+        raise UtilDateTimeValueError
+
+    return result
+
+
+def get_utc_now_datetime():
+    """Get a datetime object for utcnow()
+
+    Returns:
+          datetime: A datetime object representing utcnow().
+    """
+    return datetime.utcnow()
```

Comment:
Did you miss this comment?

Reply:
Then ensures that if we want to make a change to way forseti needs a datetime object we can do it and know it will be changed consistently throughout. It's part of the consolidation plan, otherwise we're in the scenario we were before this PR where some put use datetime.now() v. .utcnow() etc.

Reply:
I hate to say it, but this is one of those `premature optimization` people are always carping about.  :)  So I hate to press this here, but what might we possible want to change to the utcnow() object when the method name is `get_utc_now_datetime()`?